### PR TITLE
ci: support internal releases

### PR DESCRIPTION
**Pull request by tmspzz**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Our CI only builds `release` for Production. We would like to build `release/something` for internal consumption

### Solutions

Add a regex to match `release/something` branches as internal

----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1132/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4834797)(4834797)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4834797)(4834797)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1132/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4835229)(4835229)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4835229)(4835229)-dev-release.apk)

File: Jenkinsfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ def defineFlavor() {
     }
 
     def branchName = env.BRANCH_NAME
-    if (branchName == "main") {
+    if (branchName =~ /main|release\/.+/) {
```
Comment (yamilmedina) [inline Jenkinsfile:9]: 👀

**Review by yamilmedina — APPROVED** [review]
🪁

**Pull request closed without merging**